kn/fotos/api.py

```diff
@@ -141,9 +145,35 @@ def _set_metadata(data, request):
     entity.update_visibility([visibility])
     return result
 
+def _search(data, request):
+    album = entity_from_request(data)
+    if isinstance(album, basestring):
+        return {'error': album}
+
+    user = request.user if request.user.is_authenticated() else None
+    if not album.may_view(user):
+        raise PermissionDenied
+
+    if 'q' not in data:
+        return {'error': 'missing query attribute'}
+    if not isinstance(data['q'], basestring):
+        return {'error': 'query should be string'}
+    q = data['q'].strip()
+    if not q:
+        return {'error': 'query should not be empty'}
+
+    results = []
+    people = {}
+    for entity in album.search(q, user):
+        results.append(child_json(entity, user, people))
```

Comment:
Het is ongewoon in Python om argumenten op deze *pass-by-reference* wijze te gebruiken.  Wellicht kun je bij de definitie van `child_json` opmerken dat het `people` verandert.